### ISSUE TITLE
C-u next-error jumps to other file (and stays there)

### ISSUE THREAD

**Issue by simonmichael** [DamienCassou/flycheck-hledger#16]
I use ``C-u C-x` `` to start stepping through journal errors again. I expected it to restart from the top of the current file. But with two journal files open, it always jumps to one of the files, and I don't see why it picks that one. It should be reproducible with these files a.j and b.j:
```
2023-01-01
  (a)        1 = 0
```
```
2023-01-01
  (b)        1 = 0
```
It could be a flycheck bug or feature I don't understand. Can others reproduce ?

**Comment by DamienCassou** [comment]
I'm not using flycheck anymore but the builtin [Flymake](https://www.gnu.org/software/emacs/manual/html_node/emacs/Flymake.html) instead. Would you mind checking the new https://github.com/DamienCassou/flymake-hledger package as a replacement for flycheck-hledger?

**Comment by simonmichael** [comment]
Some more notes:

    C-x ` runs the command next-error (found in global-map), which is an
    interactive native-comp-function in ‘simple.el’.

    It is bound to C-x `, M-g n and M-g M-n.

    (next-error &optional ARG RESET)

    Inferred type: (function (&optional t t) (or null string))

    Visit next ‘next-error’ message and corresponding source code.

    If all the error messages parsed so far have been processed already,
    the message buffer is checked for new ones.

    A prefix ARG specifies how many error messages to move;
    negative means move back to previous error messages.
    Just C-u as a prefix means reparse the error message buffer
    and start at the first error.

    The RESET argument specifies that we should restart from the beginning.

    C-x ` normally uses the most recently started
    compilation, grep, or occur buffer.  It can also operate on any
    buffer with output from the M-x compile, M-x grep commands, or,
    more generally, on any buffer in Compilation mode or with
    Compilation Minor mode enabled, or any buffer in which
    ‘next-error-function’ is bound to an appropriate function.
    To specify use of a particular buffer for error messages, type
    C-x ` in that buffer.  You can also use the command
    ‘next-error-select-buffer’ to select the buffer to use for the subsequent
    invocation of ‘next-error’.

    Once C-x ` has chosen the buffer for error messages, it
    runs ‘next-error-hook’ with ‘run-hooks’, and stays with that buffer
    until you use it in some other buffer that uses Compilation mode
    or Compilation Minor mode.

The overlap of C-u (ARG) and RESET is a little confusing, but basically it sounds like switching to another file is not unusual, and it normally would be "the most recently started". I haven't yet figured out what logic it is using to choose the file in this flycheck-hledger case, but possibly it's working as expected.

@DamienCassou I have the impression flymake supports fewer languages and requires more manual configuration and know-how, which I am averse to right now. Once I get the familiar flycheck-hledger working again, perhaps it'll be easier for me or someone to experiment with a flymake port.

**Comment by simonmichael** [comment]
PS hopefully you are still interested in flycheck-hledger, but if not I'd be willing to take up maintenance.

**Comment by DamienCassou** [comment]
> I have the impression flymake supports fewer languages and requires more manual configuration and know-how, which I am averse to right now

I agree that flymake supports less languages but I'm not sure it requires more manual configuration. My configuration is at https://github.com/DamienCassou/emacs.d/blob/43b8e0d2b92238e8152025a8d1fa503442ac7ccc/init.el#L938 and most of it is about the mode line.

> PS hopefully you are still interested in flycheck-hledger, but if not I'd be willing to take up maintenance.

I'm still trying to review all PRs and answer all comments. But I just made you a maintainer so you don't have to wait for me anymore. Have fun.

**Comment by simonmichael** [comment]
> I'm still trying to review all PRs and answer all comments.

Thanks! I'm happy you're still maintaining flycheck-hledger for now.

> My configuration ...

Thanks, and I see your flymake-hledger configuration [here](https://github.com/DamienCassou/emacs.d/blob/43b8e0d2b92238e8152025a8d1fa503442ac7ccc/init.el#L1138-L1147). I will give flymake another try in future.

**Comment by simonmichael** [comment]
> it always jumps to one of the files

This is a design choice of Emacs' `next-error`: it "stays with that buffer until you use it in some other buffer that uses Compilation mode". Flycheck's own keybindings - C-c ! n, C-c ! p - work as I expected, staying within the current file. Closing.
